### DISCUSSION TITLE
Is the capsule concurrency configuration enforced across processes?

### DISCUSSION THREAD

**Discussion by jgmontoya** [sidekiq/sidekiq#6148] [Q&A] [answered]
I have a couple of serial queues defined with a concurrency of 1:

`config/initializers/sidekiq.rb`:
```rb
...
Sidekiq.configure_server do |config|
  ...
  (SERIAL_QUEUES).each do |serial_queue_name|
    config.capsule(serial_queue_name) do |cap|
      cap.concurrency = 1
      cap.queues = [serial_queue_name]
    end
  end
  ...
end
```

Would those queues still behave serially/non-concurrently if I start additional sidekiq processes, i.e. if I horizontally scale my sidekiq worker/server? I'm running the process by just doing `bundle exec sidekiq`. They're all using the same redis service.

Basically is the concurrency configuration enforced through redis?

**Comment by mperham** [comment, marked as answer]
Nope, capsules are per-process. The only way to limit execution across all processes is with the concurrent rate limiter in Sidekiq Enterprise.
